mat64/vector.go

```diff
@@ -357,6 +360,61 @@ func (v *Vector) MulVec(a Matrix, b *Vector) {
 	}
 }
 
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+//
+// Vector is little-endian encoded as follows:
+//   0 -  8  number of elements     (int64)
+//   8 - ..  vector's data elements (float64)
+func (v Vector) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(make([]byte, 0, sizeInt64+v.n*sizeFloat64))
+	err := binary.Write(buf, defaultEndian, int64(v.n))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, val := range v.mat.Data {
```

Comment:
Ensuring that ````v === unmarshal(marshal(v))```` depends on the internal representation. Holding that invariant either means never changing the internal representation, or means changing the binary format in the future. Neither of those seem in the spirit of BinaryMarshal. For example, in the future we may need to add a "Cap" field to Vector like we have in Dense.

I understand your concern, I just think it's not the point of BinaryMarshal. If the internal representation were to change, I would say that BinaryMarshal shouldn't change, while GobEncode should. For example, maintaining i/o equality means writing the entirety of v.mat.Data, which could be much bigger than v.Len() in the presence of inc and views.